lib/collection/variable-scope.js

```diff
@@ -214,6 +235,25 @@ _.assign(VariableScope.prototype, /** @lends VariableScope.prototype */ {
         }
 
         return obj;
+    },
+
+    /**
+     * Adds a variable list to the current instance in order to increase the surface area of variable resolution.
+     * This enables consumers to search across scopes (eg. environment and globals).
+     *
+     * @param {VariableList|Array} [definition]
+     */
+    _addLayer: function (definition) {
+        var values;
+
+        if (_.isArray(definition) || VariableList.isVariableList(definition)) {
+            definition = { values: definition };
+        }
+
+        values = definition && definition.values;
+        this.layers.push(
```

Comment:
this needs to be held by reference. so that if original scopes change, these change too.